reVX/least_cost_xmission/cost_creator.py

```diff
@@ -0,0 +1,410 @@
+
+        if 'slope' in d_mults:
+            r_slope = slope[default_mask]
+            slope_mult = self._compute_slope_mult(r_slope, d_mults['slope'])
+            mults_arr[default_mask] = mults_arr[default_mask] * slope_mult
+
+        # Set water multiplier last so we don't get super high multipliers at
+        # water body boundaries next to steep slopes
+        mults_arr[land_use == WATER_NLCD_CODE] = WATER_MULT
+
+        return mults_arr
+
+    def compute_base_line_costs(self, capacity, base_line_costs, cell_size):
+        """
+        Get base line cost per cell raster for a given voltage
+
+        Parameters
+        ----------
+        capacity : int
+            Desired line capacity (MW). This must be in base_line_costs.
+        base_line_costs : dict
+            Base line cost per mile for ISOs
+        cell_size : int
+            Raster cell size for all layers (meters). Assumed to be square.
+
+        Returns
+        -------
+        base_cost : numpy.ndarray
+            Cost per cell raster in same shape as iso_regions
+        """
+        base_cost = np.full(self._iso_regions.shape, float('inf'),
+                            dtype=np.float32)
+
+        for iso in base_line_costs:
+            logger.info(f'Processing costs for {iso} for {capacity}MW')
+            iso_code = self._iso_lookup[iso]
+            cost_per_mile = base_line_costs[iso][str(capacity)]
+            cost_per_cell = cost_per_mile / METERS_IN_MILE * cell_size
+            logger.debug(f'$/mile is {cost_per_mile}, $/cell is '
+                         f'{cost_per_cell}')
+            mask = self._iso_regions == iso_code
+            base_cost[mask] = cost_per_cell
+
+        return base_cost
+
+    def save_layer(self, layer_name, data):
+        """
+        Save data to h5 file
+
+        Parameters
+        ----------
+        layer_name : str
+            Name of layer to save in h5
+        data : np.ndarray
+            Data to savein h5
+        """
+        logger.debug(f'Saving {layer_name} to h5')
+        self._profile['nodata'] = None
+        self._write_layer(self._excl_h5, layer_name, self._profile, data)
+
+    def create_geotiff(self, tiff_dir, geotiff_fpath, data):
+        """
+        Save geotiff to disk
+
+        Parameters
+        ----------
+        tiff_dir : str
+            Directory to save tiffs in
+        geotiff_fpath : str
+            File name of tiff including extension
+        data : np.ndarray
+            Data to save
+        """
+        nodata = self._profile['nodata']
+        if tiff_dir != '':
+            geotiff_fpath = os.path.join(tiff_dir, geotiff_fpath)
+        logger.debug(f'Saving {geotiff_fpath}')
+        self._write_geotiff(geotiff_fpath, self._profile, data)
+
+        # _write_geotiff sets nodata to the max for the dtype. For float32 this
+        # then breaks JSONifying the profile when writing h5s. Reset it.
+        self._profile['nodata'] = nodata
```

Comment:
Why is nodata pulled out and then added back to the profiel? I dont' see it being used in this function, are the above comments supposed to be a TODO?